dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -708,15 +710,33 @@ public Tool getContainerByToolPath(@ApiParam(hidden = true) @Auth User user,
     @Path("/path/tool/{repository}/published")
     @ApiOperation(value = "Get a published tool by the specific tool path.", notes = "Requires full path (including tool name if applicable).", response = Tool.class)
     public Tool getPublishedContainerByToolPath(
-        @ApiParam(value = "repository path", required = true) @PathParam("repository") String path, @ApiParam(value = "Comma-delimited list of fields to include: validations") @QueryParam("include") String include) {
+        @ApiParam(value = "repository path", required = true) @PathParam("repository") String path, @ApiParam(value = "Comma-delimited list of fields to include: validations") @QueryParam("include") String include, @Context SecurityContext securityContext, @Context ContainerRequestContext containerContext) {
         try {
             Tool tool = toolDAO.findByPath(path, true);
             checkEntry(tool);
 
             if (checkIncludes(include, "validations")) {
-                tool.getTags().forEach(tag -> Hibernate.initialize(tag.getValidations()));
+                tool.getWorkflowVersions().forEach(tag -> Hibernate.initialize(tag.getValidations()));
             }
             filterContainersForHiddenTags(tool);
+
+            // for backwards compatibility for 1.6.0 clients, return versions as tags
+            // this seems sufficient to maintain backwards compatibility for launching
+            try {
+                final List<String> strings = containerContext.getHeaders().get("User-Agent");
+                strings.forEach(s -> {
```

Comment:
Probably null. 
I was being a bit lazy here with the catch since I expect that this code should be deleted shortly after release when we've confirmed that most have migrated to the newer CLI (since we know via #2355 and co. that there is no cloud launch-with tools).